### PR TITLE
Move IP Adapter Scripts to research project

### PR DESCRIPTION
**Pull request by ParagEkbote**
# What does this PR do?

I have moved the IP Adapter Scripts to the research project folder as mentioned, to close #7196. Please let me know if any further corrections are needed and I will make the necessary changes.

Similar to #9935 .

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).

## Who can review?

@sayakpaul

### CODE REVIEW

**Comment by sayakpaul** [comment]
Can you cherry-pick the commits from https://github.com/huggingface/diffusers/pull/7196 and then add a commit to move them to research folder? This way, the original contributions receive the proper credits they deserve.

**Comment by pcuenca** [comment]
> Can you cherry-pick the commits from #7196 and then add a commit to move them to research folder?

Agreed, or start the branch from the other PR's branch if that's easier.

**Comment by ParagEkbote** [comment]
I have copied the contents from #7196 and added them to research project. Should I credit the user in the ReadME, since the branch in that repository is out-of-date. 

cc: @sayakpaul

**Comment by sayakpaul** [comment]
You should still be able to cherry-pick commits.

**Comment by ParagEkbote** [comment]
> You should still be able to cherry-pick commits.

I have tried a couple of times, but unable to create a PR due to repository permission conflicts.

**Comment by sayakpaul** [comment]
Can you run `make style && make quality`?

File: docs/source/en/_toctree.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@
     title: Working with big models
   title: Tutorials
 - sections:
+<<<<<<< HEAD
```
Comment (pcuenca) [inline docs/source/en/_toctree.yml:28]: These are git conflict markers that need to be resolved.

**Comment by ParagEkbote** [comment]
I have run the styling checks and resolved the git conflict. Please let me know if any further corrections are needed and I will make the necessary changes.

cc: @pcuenca

File: docs/source/en/_toctree.yml [hunk 2]
```diff
@@ -25,19 +25,135 @@
...
-    title: Load community pipelines and components
-  - local: using-diffusers/schedulers
-    title: Load schedulers and models
-  - local: using-diffusers/other-formats
-    title: Model files and layouts
-  - local: using-diffusers/loading_adapters
-    title: Load adapters
-  - local: using-diffusers/push_to_hub
-    title: Push files to the Hub
-  title: Load pipelines and adapters
```
Comment (sayakpaul) [inline docs/source/en/_toctree.yml:40]: We should not be changing this

File: docs/source/en/training/ip_adapter.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+# IP Adapter Training Example
```
[the remark this answers is not in the document]
Comment (ParagEkbote) [reply docs/source/en/training/ip_adapter.md]: > I think this can be moved to the README.md as well if there's any difference in the content. This is because for projects within `research_projects` folder, we typically don't include a doc

The files are same, so I removed the file.

File: docs/source/en/_toctree.yml (hunk 2 above)
Comment (ParagEkbote) [reply docs/source/en/_toctree.yml]: I have updated toctree.yml to reset the changes.

File: docs/source/en/_toctree.yml [hunk 4]
```diff
@@ -569,4 +569,4 @@
     - local: api/video_processor
       title: Video Processor
     title: Internal classes
-  title: API
+  title: API
```
Comment (sayakpaul) [inline docs/source/en/_toctree.yml:572]: No change here.

**Comment by ParagEkbote** [comment]
@sayakpaul Sorry, I'll try to do better.

File: docs/source/en/_toctree.yml [hunk 5]
```diff
@@ -568,5 +568,6 @@
       title: VAE Image Processor
     - local: api/video_processor
       title: Video Processor
-    title: Internal classes
+      title: Internal classes
   title: API
+
```
Comment (sayakpaul) [inline docs/source/en/_toctree.yml:573]: Why this change?

Comment (ParagEkbote) [reply docs/source/en/_toctree.yml:573]: I have referenced the toctree.yml file from the main branch. In previous commits, the CI tests kept failing, so I corrected it.

**Comment by sayakpaul** [comment]
@stevhliu do the docs-related changes look good to you?

File: docs/source/en/_toctree.yml (hunk 5 above)
Comment (a-r-r-o-w) [reply docs/source/en/_toctree.yml:573]: Please run `git restore -s main /path/to/toctree.yaml` and push the changes

Comment (stevhliu) [reply docs/source/en/_toctree.yml:573]: This should be restored to how it was previously. We can help fix the CI test if it fails

```yaml
    - local: api/video_processor
      title: Video Processor
    title: Internal classes
```

**Review by stevhliu — COMMENTED** [review]
I noticed there are also some unresolved comments in the last PR [here](https://github.com/huggingface/diffusers/pull/7196/files#r1510551508) and [here](https://github.com/huggingface/diffusers/pull/7196/files#r1510551762). Do you think you could also resolve those comments?

**Review by stevhliu — APPROVED** [review]
Thanks!

**Comment by ParagEkbote** [comment]
Could you please see that #7196 gets closed to declutter?

cc: @stevhliu
